pkg/bootstrap/server_v2.go

```diff
@@ -136,7 +136,12 @@ func (h *ServerV2) Auth(ctx context.Context, authReq *bootstrapv2.BootstrapAuthR
 	kr := keyring.New(keyring.NewSharedKeys(sharedSecret))
 
 	tokenLabels := maps.Clone(bootstrapToken.GetMetadata().GetLabels())
-	delete(tokenLabels, corev1.NameLabel)
+
+	// if the token is not a one-time token, remove the name label
+	if bootstrapToken.GetMetadata().GetMaxUsages() != 1 {
+		delete(tokenLabels, corev1.NameLabel)
+	}
```

Comment:
```suggestion
	// if the token is not a one-time token, remove the name label
	if bootstrapToken.GetMetadata().GetMaxUsages() - bootstrapToken.GetMetadata().GetUsageCount() > 1 {
		delete(tokenLabels, corev1.NameLabel)
	}
```

Reply:
MaxUsages of 0 is unlimited usages so this change will not be valid.

Reply:
resolved offline, original logic is correct